test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -1102,6 +1176,79 @@ private static IEdmModel GetEdmModel()
             }
         }
 
+        /// <summary>
+        /// Write a message using an <see cref="ODataMessageWriter"/> instance
+        /// and returns the written payload as a string.
+        /// </summary>
+        /// <param name="edmModel">The <see cref="IEdmModel"/> used to initialize the writer.</param>
+        /// <param name="UseLegacyScaleAndSridVariable">Whether to use legacy scale and srid variable casing.</param>
+        /// <param name="test">The action that writes the payload.</param>
+        /// <param name="message">
+        /// The message instance that encapsulate the request.
+        /// If none is provided, a new instance of <see cref="InMemoryMessage"/>.
+        /// The provided instance might be modified by this method.
+        /// </param>
+        /// <param name="encoding">
+        /// The encoding to use when reading from the request stream. This defaults to UTF8.
+        /// If you set a custom <paramref name="contentType"/>, you should specify a matching encoding.
+        /// </param>
+        /// <param name="configureServices">
+        /// Action to inject services to the dependency-injection container.
+        /// If this is set, then the generated <see cref="IServiceProvider"/> will be added to the <paramref name="message"/>.Container property.
+        /// </param>
+        /// <param name="path">
+        /// The OData Uri path.
+        /// </param>
+        /// <returns>The written output.</returns>
+        private string WriteAndGetPayloadForScaleAndSridVariables(
+            IEdmModel edmModel,
+            bool useLegacyScaleAndSridVariable,
+            Action<ODataMessageWriter> test,
+            InMemoryMessage message = null,
+            Encoding encoding = null,
+            Action<IServiceCollection> configureServices = null,
+            ODataPath path = null)
+        {
+            message = message ?? new InMemoryMessage() { Stream = new MemoryStream() };
+
+            if (configureServices != null)
+            {
+                IServiceProvider container = CreateTestServiceContainer(configureServices);
+                message.ServiceProvider = container;

```

Comment:
When initializing the `ODataMessageWriter`, you cast `InMemoryMessage` to `IODataResponseMessage`. The interface knows nothing about `ServiceProvider`. Does `ServiceProvider` ever get used?

Reply:
I've removed the serviceprovider and path but because the stream is used..I've left the InMemoryMessage.